tracecontext.go

```diff
@@ -169,23 +169,61 @@ type TraceState struct {
 
 // NewTraceState returns a TraceState based on entries.
 func NewTraceState(entries ...TraceStateEntry) TraceState {
-	out := TraceState{}
+	var out TraceState
 	var last *TraceStateEntry
+	var modified []TraceStateEntry
+	recorded := make(map[string]uint8)
```

Comment:
WDYT about not checking duplicates for non-Elastic keys? We don't really care about them. If we're not required to check for duplicates I think it might be better to only check our own keys, as we can do it without any allocations: just keep track of whether we've seen an Elastic key already.

Reply:
that's probably better than treating all the keys the same, since each system should do their own key _cleaning_. I'll change the implementation and simplify the logic.